### PR TITLE
Fix inconsistencies in polynomial autograd tutorial

### LINKED ISSUE #3632: Feedback about PyTorch: Tensors and autograd
There is the following issue on this page: https://docs.pytorch.org/tutorials/beginner/examples_autograd/polynomial_autograd.html

The paratext and previous/subsequent excercises imply this should learn sin(x), but the y= is generated as torch.exp(x), with the exponential's Taylor expantion.

### PR DESCRIPTION
**Pull request by gtsitsik**
## Changes

- Use sine instead of exponential to match the tutorial text.
- Update the learning rate and iteration count to match `beginner_source/examples_tensor/polynomial_tensor.py` which is exactly the same tutorial but with manual gradient computations.
- Correct the description of the `x.grad` attribute.
- Remove redundant lines of code.
- Fix typo. 

## Note
PR #3663 and PR #3897 recommended updating the tutorial text to show the exponential function instead of updating the code to use a sine. However, I recommend using sine everywhere since both the numpy and non-autograd pytorch polynomial tutorials use a sine. 

Fixes #3632 

## Checklist

- [x] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

### CODE REVIEW

**Comment by telamonian** [comment]
Dueling fixes (full disclosure I'm the author of #3897)!

@gtsitsik I don't know if you saw #3597, the PR that originally (and fairly recently) changed the tutorial from using sin(x) to e^x, but they gave a pretty reason for doing so. Basically, the trained Taylor expansion given in the tutorial is able to converge nicely to e^x, but does not converge that nicely to sin(x). You can see the problem in the following plots:

Untitled

Untitled-1

The problem is fundamentally a mathematical one; a third order polynomial is never a good fit for sin(x), while at least over the domain [-1, 1] a third order polynomial is a good fit for e^x.

Therefore I propose that Pytorch keeps the tutorial using e^x while also keeping @gtsitsik's fixes of typos and removal of unused code. I've updated #3897 with these fixes.

**Comment by gtsitsik** [comment]
@telamonian by the way I noticed your PR after I created mine.

Regarding using an exponential to get a better approximation, I don't think it really matters for the purposes of this tutorial. But I also don't mind if it remains an exponential. 

However, I do think that the 3 tutorials should be consistent with each other as they seem to be aiming to demonstrate the same thing, but via different approches. That is, 

- manual gradient calculation using numpy 
- manual gradient calculation using pytorch 
- automatic gradient calculation using pytorch 

All 3 should either use a sine or an exponential because otherwise they will be inconsistent with each other and their pedagogical quality becomes slightly off.

**Comment by telamonian** [comment]
> All 3 should either use a sine or an exponential because otherwise they will be inconsistent with each other

That I can definitely agree with. But it doesn't all have to be changed in one big PR. I think the best thing to do for now would be to pull in #3897, close this PR, and then fix the equations/descriptions for the other examples in future PRs.

But we'll have to wait and see what any of the actual maintainers think

**Comment by gtsitsik** [comment]
@telamonian that is fine with me. However, in my PR I include another fix that you didn't include in yours. Specifically, the tutorial incorrectly claims that `grad.x` holds the gradient of `x` with respect to some scalar (the loss), when in reality it is the gradient of the loss with respect to `x`.
